src/doc_builder/commands/notebook_to_mdx.py

```diff
@@ -14,60 +14,44 @@
 # limitations under the License.
 
 import argparse
+import subprocess
+from pathlib import Path
 
-import nbformat
-
-from ..style_doc import format_code_example
+from tqdm import tqdm
 
 
 def notebook_to_mdx_command(args):
-    notebook = nbformat.read(args.notebook_file, as_version=4)
-    content = []
-    for cell in notebook["cells"]:
-        if cell["cell_type"] == "code":
-            code = cell["source"]
-            outputs = [
-                o for o in cell["outputs"] if ("text" in o and o.get("name", None) == "stdout") or "text/plain" in o
-            ]
-            if len(outputs) > 0:
-                code_lines = code.split("\n")
-                # We can add >>> everywhere without worrying as format_code_examples will replace them by ...
-                # when needed.
-                code_lines = [f">>> {l}" if not len(l) == 0 or l.isspace() else l for l in code_lines]
-                code = "\n".join(code_lines)
-                code = format_code_example(code, max_len=args.max_len)[0]
-                content.append(f"```python\n{code}\n```")
-
-                output = outputs[0]["text"] if "text" in outputs[0] else outputs[0]["text/plain"]
-                output = output.strip()
-                content.append(f"```python out\n{output}\n```")
-            else:
-                code = format_code_example(code, max_len=args.max_len)[0]
-                content.append(f"```python\n{code}\n```")
-        elif cell["cell_type"] == "markdown":
-            content.append(cell["source"])
-        else:
-            content.append(f"```\n{cell['source']}\n```")
-
-    dest_file = args.dest_file if args.dest_file is not None else args.notebook_file.replace(".ipynb", ".mdx")
-    with open(dest_file, "w", encoding="utf-8") as f:
-        f.write("\n\n".join(content))
+    src_path = Path(args.notebook_src).resolve()
+    src_dir = src_path.parent if src_path.is_file() else src_path
+    notebook_paths = [str(src_path)] if src_path.is_file() else [*src_dir.glob("*.ipynb")]
+    output_dir = src_dir if args.output_dir is None else Path(args.output_dir).resolve()
+
+    try:
+        for notebook_path in tqdm(notebook_paths, desc="Converting .ipynb files to .md files"):
+            subprocess.run(
+                ["jupyter", "nbconvert", notebook_path, "--to", "markdown", "--output-dir", output_dir],
```

Comment:
not sure if it makes sense/if it's relevant but just noting that we have an internal nbconvert API here: https://huggingface.co/spaces/julien-c/nbconvert

Reply:
good to know. But right now I think it is fine to use `nbconvert` locally rather than through API